doc/customizing_guide/toolbar.rst

```diff
@@ -0,0 +1,205 @@
+Custom tools for viewers and custom toolbars
+============================================
+
+Writing a custom tool for a viewer toolbar
+------------------------------------------
+
+Here we take a look at how to create a tool to include in a viewer's toolbar
+(either one of the built-in viewers or a custom viewer) There are two types of
+tools: ones that can be checked and unchecked, and ones that simply trigger an
+event when pressed, but do not remain pressed. These are described in the
+following two sub-sections.
+
+Non-checkable tools
+^^^^^^^^^^^^^^^^^^^
+
+The basic structure for a non-checkable tool is:
+
+.. code:: python
+
+    from glue.config import viewer_tool
+    from glue.viewers.common.qt.tool import Tool
+
+    @viewer_tool
+    class MyCustomTool(Tool):
+
+        icon = 'myicon.png'
+        tool_id = 'custom_tool'
+        action_text = 'Does cool stuff'
+        tool_tip = 'Does cool stuff'
+        shortcut = 'D'
+
+        def __init__(self, viewer):
+            super(MyCustomMode, self).__init__(viewer)
+
+        def activate(self):
+            pass
+
+        def close(self):
+            pass
+
+The class-level variables set at the start of the class are as follows:
+
+* ``icon``: this should be set either to the name of a built-in glue icon, or
+  to the path to a PNG file to be used for the icon. Note that this should
+  **not** be a ``QIcon`` object.
+
+* ``tool_id``: a unique string that identifies this tool. If you create a
+  tool that has the same ``tool_id`` as an existing tool already implemented in
+  glue, you will overwrite the existing tool. This is never shown to the user.
+
+* ``action_text``: a string describing the tool. This is shown in the status bar
+  at the bottom of the viewer whenever the button is active.
+
+* ``tool_tip``: this should be a string that will be shown when the user hovers
+  above the button in the toolbar. This can include instructions on how to use
+  the tool.
+
+* ``shortcut``: this should be a string giving a key that the user can press
+  when the viewer is active, which will activate the tool. This can include
+  modifier keys, e.g. ``'Ctrl+A'`` or ``'Ctrl+Shift+U'``, but can also just be
+  a single key, e.g. ``'K'``.
+
+When the user presses the tool icon, the ``activate`` method is called. In this
+method, you can write any code including code that may for example open a Qt
+window, or change the state of the viewer (for example changing the zoom or
+field of view). You can access the viewer instance with ``self.viewer``.
+Finally, when the viewer is closed the ``close`` method is called, so you should
+use this to do any necessary cleanup.
+
+The ``@viewer_tool`` decorator tells glue that this class represents a viewer
```

Comment:
Personally, I like to either capitalize a package name or to mark it like `glue`, so it is obvious you are referring to the package, and not the sticky kind. But please ignore if you disagree.